common/src/main/java/io/pravega/common/util/Retry.java

```diff
@@ -206,7 +210,39 @@ private RetryAndThrowBase(Class<ThrowsT> throwType, RetryWithBackoff params) {
             }
             throw new RetriesExhaustedException(last);
         }
-
+        
+        public CompletableFuture<Void> runInExecutor(final Runnable task,
+                                                     final ScheduledExecutorService executorService) {
+            Preconditions.checkNotNull(task);
+            CompletableFuture<Void> result = new CompletableFuture<>();
```

Comment:
Yes method, `io.pravega.common.util.Retry.RetryAndThrowBase#runInExecutor` and `io.pravega.common.util.Retry.RetryAndThrowBase#runAsync` can remove this unnecessary CompletableFuture<Void> result.  Raising an issue for it  #1660